### ISSUE TITLE
[3.0] Support CardHolderName

### ISSUE THREAD

**Issue by judgej** [thephpleague/omnipay-common#129]
Most gateways now distinguish between the billing name and the card holder name. This is useful when, for example, a person is being billed but is using a company credit card - the names will be different.

The `CreditCard` class does not explicitly provide a named field for the card holder name. For the sake of standardisation and setting a convention that all the drivers can use, I believe it needs a card holder name parameter, whatever it is called.

The card holder name *could* fall back to the billing name if it is not explicitly set, so this would provide some backwards compatibility.

Any thoughts on that? Good idea? Perhaps I'm overlooking something?

**Comment by barryvdh** [comment]
I guess so. I would also prefer to remove the Customer object from the CreditCard object entirely.

**Comment by judgej** [comment]
Yes, I'd be behind that. The credit card is just one of a range of payment methods. Many gateways support many other methods, all with different collections of data to define them: previous transactions, saved card references (with or without a CVV grabbed from the user last-minute for extra security), short-lived tokenised cards, direct debit mandate, bank accounts, PayPal tokens etc. The credit card payment method is not special - it only needs half a dozen pieces of data to define an entered credit card.

**Comment by judgej** [comment]
It was this ticket that prompted this: https://github.com/thephpleague/omnipay-sagepay/issues/81
